### ISSUE TITLE
Closing inspector throws exception after merging #5612

### ISSUE THREAD

**Issue by bionoid** [kivy/kivy#5645]
### Versions

* Python: 3.6.4
* OS: Linux
* Kivy: master

### Description

Closing inspector throws an exception after merging #5612. Seems like the issue is that [when calling remove_widget here](https://github.com/kivy/kivy/blob/1.10.0/kivy/modules/inspector.py#L381), removing from children list (which triggers this callback) happens before clearing .parent property

Edit to add: The exception is caught, doesn't crash the application

### Code and Logs

`python any.py -m inspector`, press ctrl-e and hit "x" to close the popup, result:

```
 Traceback (most recent call last):
   File "kivy/properties.pyx", line 571, in kivy.properties.Property.dispatch
   File "kivy/_event.pyx", line 1214, in kivy._event.EventObservers.dispatch
   File "kivy/_event.pyx", line 1120, in kivy._event.EventObservers._dispatch
   File "/home/kivy/kivy/kivy/modules/inspector.py", line 382, in on_window_children
     win.add_widget(self)
   File "/home/kivy/kivy/kivy/core/window/__init__.py", line 1247, in add_widget
     (widget, widget.parent)
 kivy.uix.widget.WidgetException: Cannot add <kivy.modules.inspector.Inspector object at 0x7f5f8645f1e8> to window, it already has a parent <kivy.core.window.window_sdl2.WindowSDL object at 0x7f5f9654a730>
 Exception ignored in: 'kivy.properties.observable_list_dispatch'
 Traceback (most recent call last):
   File "kivy/properties.pyx", line 571, in kivy.properties.Property.dispatch
   File "kivy/_event.pyx", line 1214, in kivy._event.EventObservers.dispatch
   File "kivy/_event.pyx", line 1120, in kivy._event.EventObservers._dispatch
   File "/home/kivy/kivy/kivy/modules/inspector.py", line 382, in on_window_children
     win.add_widget(self)
   File "/home/kivy/kivy/kivy/core/window/__init__.py", line 1247, in add_widget
     (widget, widget.parent)
 kivy.uix.widget.WidgetException: Cannot add <kivy.modules.inspector.Inspector object at 0x7f5f8645f1e8> to window, it already has a parent <kivy.core.window.window_sdl2.WindowSDL object at 0x7f5f9654a730>
```

**Comment by KeyWeeUsr** [comment]
The same thing makes some tests in CI fail.

**Comment by tshirtman** [comment]
apparently this merge didn't fix the CI builds (currently it failed on 2.7 and 3.5 on a stuck inspector test).

**Comment by Zen-CODE** [comment]
Well, this error was fixed by this commit: https://github.com/kivy/kivy/commit/1531a0c984a22b8a447c68cf5ca45ea7e58b74e2

But the build still fails as the "test_widget_popup" call never exists in "test_module_inpector.py" for some reason. Possibly related, but I could not see why that happens. For a full discussion:

https://github.com/kivy/kivy/pull/5658#issuecomment-381732636

But this error was fixed. So lets close it.
